subprojects/docs/src/docs/userguide/moreAboutTasks.adoc

```diff
@@ -580,6 +580,49 @@ Input properties annotated with `@api:org.gradle.api.tasks.CompileClasspath[]` a
 Additionally to the aforementioned general classpath rules, compile classpaths ignore changes to everything but class files. Gradle uses the same class analysis described in <<sec:java_compile_avoidance>> to further filter changes that don't affect the class' ABIs.
 This means that changes which only touch the implementation of classes do not make the task out of date.
 
+[[sec:task_input_nested_inputs]]
+====== Nested inputs
+
+When analyzing `@api:org.gradle.api.tasks.Nested[]` task properties for declared input and output sub-properties Gradle uses the type of the actual value.
+Hence it can discover all sub-properties declared by a runtime sub-type.
+
+When adding `@api:org.gradle.api.tasks.Nested[]` to an iterable, each element is treated as a separate nested input.
+
+This allows some richer modelling and extensibility for tasks, as e.g. shown by `api:org.gradle.api.tasks.compile.CompileOptions#getCompilerArgumentProviders[]`.
```

Comment:
Typos: either "This allows for some richer modeling..." or "This allows richer modeling", in either case "modeling" has a single "l".

Reply:
fixed.